### ISSUE TITLE
Unrelated error when adding an ECS cluster via aws_ecs_service without network configuration

### ISSUE THREAD

**Issue by timello** [hashicorp/terraform#23920]
### Terraform Version

```
terraform --version
Terraform v0.12.19
+ provider.aws v2.45.0
```

### Terraform Configuration Files

```hcl
terraform {
  required_version = ">= 0.12.0"
}

provider "aws" {
  region = "eu-central-1"
}

resource "aws_ecs_cluster" "example_ecs_cluster" {
  name = "example"
}

resource "aws_ecs_task_definition" "example_task_definition" {
  family                   = "example"
  network_mode             = "awsvpc"
  requires_compatibilities = ["FARGATE"]
  cpu = 256
  memory = 512

  container_definitions = <<DEFINITION
[
  {
    "name": "myapp",
    "image": "nginx",
    "essential": true,
    "portMappings": [
      {
        "containerPort": 8080,
        "hostPort": 8080
      }
    ]
  }
]
DEFINITION
}

resource "aws_ecs_service" "example_ecs_service" {
  name            = "example_service"
  launch_type     = "FARGATE"
  cluster         = aws_ecs_cluster.example_ecs_cluster.arn
  task_definition = aws_ecs_task_definition.example_task_definition.arn
  desired_count   = 1
}
```

### Debug Output

https://gist.github.com/timello/290d9c13f7d864d5761313a3f8e2ab56

### Expected Behavior

Terraform should return an error saying that the "network_configuration" is required since "network_mode" is "awsvpc".
### Actual Behavior

Terraform returns a completely unrelated error message as shown below:
```
Error: InvalidParameterException:  "example_service"

  on main.tf line 37, in resource "aws_ecs_service" "example_ecs_service":
  37: resource "aws_ecs_service" "example_ecs_service" {
```
### Steps to Reproduce

1. `terraform init`
2. `terraform apply`

**Comment by jbardin** [comment]
Hi @timello,

The error `Error: InvalidParameterException:  "example_service"` is coming from the AWS api, not from the AWS plugin or Terraform itself. Unfortunately it doesn't seem to have much info, but appears that the name is somehow invalid, possibly a duplicate?

Since there is no additional information returned to Terraform about the error, there isn't a way to present anything more precise in the UI here. If this is somehow related to the missing parameter, and this config and environment is otherwise valid, please file an issue with the AWS provider, including the relevant trace logs which would include AWS API calls. 

Thanks

**Comment by guidoffm** [comment]
@jbardin I am afrais this information is not correct. The cause is shown in the trace file. I have opened #23941 for this.

**Comment by bflad** [comment]
To followup here from the provider side, this appears very similar to a regression in version 2.45.0 of the Terraform AWS Provider (https://github.com/terraform-providers/terraform-provider-aws/issues/11682) and was resolved in [Terraform AWS Provider version 2.46.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2460-january-23-2020), which was released January 23rd. Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading. 👍
